Assets/MixedRealityToolkit/_Core/Devices/BaseController.cs

```diff
@@ -114,7 +114,15 @@ public bool SetupConfiguration(Type controllerType)
                         controllerMappings[i].Handedness == ControllerHandedness &&
                         controllerMappings[i].Interactions.Length > 0)
                     {
-                        AssignControllerMappings(controllerMappings[i].Interactions);
+                        MixedRealityInteractionMapping[] profileInteractions = controllerMappings[i].Interactions;
+                        MixedRealityInteractionMapping[] newInteractions = new MixedRealityInteractionMapping[profileInteractions.Length];
+
+                        for (int j = 0; j < profileInteractions.Length; j++)
+                        {
+                            newInteractions[j] = new MixedRealityInteractionMapping(profileInteractions[j]);
```

Comment:
Wouldn't this duplicate mappings for every controller?

@SimonDarksideJ thoughts?

Reply:
Yep, it would.

As mentioned in the overview above:

>This will also fix cases where a source might be lost while selecting, which would cause the state to change when a second, unselecting source is detected.

Reply:
A generic solution like this will fix the problem for all mappings that might be reused, like the touch mappings which appear to be reused for each individual touch.

Reply:
Oh, I see, it's only duplicating it for the specific controller being configured, correct?

Reply:
So if someone was to have two touch controllers be spawned then each would make it's own copy to use, do I have that right?

Reply:
You have that exactly right. This happens in the setup of individual controllers.

What was happening before was that both HoloLens hand controllers (or touch controllers, etc) were referencing the same single array from the profile, so their values were fighting and overwriting each other.

Reply:
I can see why that'd be a problem, lol